treetop.js

```diff
@@ -325,6 +418,28 @@ window.treetop = (function ($, BodyComponent, FormSerializer) {
     PARTIAL_CONTENT_TYPE: "application/x.treetop-html-partial+xml",
     FRAGMENT_CONTENT_TYPE: "application/x.treetop-html-fragment+xml",
 
+    START: "treetopstart",
+    COMPLETE: "treetopcomplete",
+
+    startRequest: function () {
+        this.activeCount++;
+        if (this.activeCount === 1) {
+            var event = document.createEvent("Event");
+            event.initEvent(this.START, false, false);
+            document.dispatchEvent(event);
+        }
+    },
+
+    endRequest: function () {
+        this.activeCount--;
+        if (this.activeCount < 1) {
+            this.activeCount = 0;
+            var event = document.createEvent("Event");
+            event.initEvent(this.COMPLETE, false, false);
+            document.dispatchEvent(event);
```

Comment:
Do we need to create a new event each time or can it be created once and dispatched multiple times?

Reply:
Interesting question, might be able to reuse, although seems dubious somehow. The dispatch of this event is throttled by XHR requests so this is fairly unlikely to be the bottleneck in case of runaway code.